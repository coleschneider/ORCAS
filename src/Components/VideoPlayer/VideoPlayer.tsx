import * as React from 'react';
import Loader from 'Common/Loader/Loader';
import ReactPlayer from 'react-player';

interface VideoPlayerState {
  isFetching: boolean;
}
class VideoPlayer extends React.Component<{}, VideoPlayerState> {
  constructor(props) {
    super(props);
    this.state = {
      isFetching: true,
    };
  }
  render() {
    const { isFetching } = this.state;

    return (
      <React.Fragment>
        {isFetching && <Loader isFetching={isFetching} />}
        <ReactPlayer
          width="100%"
          onReady={() => this.setState({ isFetching: false })}
          onError={() => this.setState({ isFetching: false })}
          style={{ display: isFetching ? 'none' : undefined }}
          height="100%"
          url="https://player.vimeo.com/video/223376401?color=00a650&title=0&byline=0&portrait=0"
          config={{
            vimeo: {
              playerOptions: {
                height: 200,
              },
            },
          }}
        />
      </React.Fragment>
    );
  }
}

export default VideoPlayer;
