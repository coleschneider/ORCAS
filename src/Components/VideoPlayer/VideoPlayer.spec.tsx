import * as React from 'react';
import VideoPlayer from './VideoPlayer';

describe('VideoPlayer', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<VideoPlayer />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should display the spinner and set state to fetching', () => {
    const wrapper = shallow(<VideoPlayer />);
    wrapper
      .find('ReactPlayer')
      .props()
      .onReady();
    expect(wrapper.state().isFetching).toBe(false);
  });
});
