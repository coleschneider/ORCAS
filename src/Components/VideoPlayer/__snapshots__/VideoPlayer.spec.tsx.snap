// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoPlayer should match the snapshot 1`] = `
<Fragment>
  <Loader
    isFetching={true}
  />
  <ReactPlayer
    config={
      Object {
        "vimeo": Object {
          "playerOptions": Object {
            "height": 200,
          },
        },
      }
    }
    controls={false}
    height="100%"
    light={false}
    loop={false}
    muted={false}
    onBuffer={[Function]}
    onBufferEnd={[Function]}
    onDisablePIP={[Function]}
    onDuration={[Function]}
    onEnablePIP={[Function]}
    onEnded={[Function]}
    onError={[Function]}
    onPause={[Function]}
    onPlay={[Function]}
    onProgress={[Function]}
    onReady={[Function]}
    onSeek={[Function]}
    onStart={[Function]}
    pip={false}
    playbackRate={1}
    playing={false}
    playsinline={false}
    progressInterval={1000}
    style={
      Object {
        "display": "none",
      }
    }
    url="https://player.vimeo.com/video/223376401?color=00a650&title=0&byline=0&portrait=0"
    volume={null}
    width="100%"
    wrapper="div"
  />
</Fragment>
`;
